### DISCUSSION TITLE
OAuth 404 on iOS Safari (missing subscription client)

### DISCUSSION THREAD

**Discussion by j0sh** [pocketbase/pocketbase#3766] [Q&A]
Hi - I am hitting the following issue with the all-in-one OAuth flow on iOS Safari (with the [workaround](https://github.com/pocketbase/pocketbase/discussions/2429#discussioncomment-5943061)).

Often I am seeing a 404 error from Pocketbase when trying to login via OAuth (Github), with the error message "Missing or invalid OAuth2 subscription client"

From what I can infer: the all-in-one flow starts up a new tab for the OAuth provider, and the app's own login tab is put into the background. If authentication takes more than a few seconds (say, the user has to dig up their password and/or 2fa credential for the provider), then iOS will close the realtime connection in the background tab. When OAuth completes, Pocketbase tries to call back to a non-existent connection, which fails with the 404 error.

Is this an accurate interpretation of the issue, or might something else be happening here? Are there any workarounds?

Example logs below:

```
2023/11/16 18:49:09 Realtime connection established: 3JM3shv0siTZSSerVrpdWIUruVaRUYk0xorj9TC4
2023/11/16 18:50:04 Realtime connection closed (cancelled request): 3JM3shv0siTZSSerVrpdWIUruVaRUYk0xorj9TC4
2023/11/16 18:50:21 Realtime connection established: cHX7F3kZjYAwK1q2J9PjLxsjH21WVi0LaYlO49po
2023/11/16 18:50:40 No client associated with connection ID "3JM3shv0siTZSSerVrpdWIUruVaRUYk0xorj9TC4"
```

**Comment by ganigeorgiev** [comment]
The "all-in-one" flow relies that the SSE connection will remain active during the OAuth2 authentication.

If iOS is closing the connection in the background tab then Yes, it is possible the API to return 404.

Unfortunately I don't have Mac or iOS device to test it and I'm not sure I'll be able to help you with this.

The only workaround I can suggest is to use the "Manual code exchange" flow.

_I'll move the issue to Q&A discussion._

**Reply by ganigeorgiev** [reply]
_Side-note: depending how your app is deployed, it is also possible the realtime connection to be interrupted by your host/reverse-proxy (_fly.io_ for example I remember having ~60s limit)._

**Comment by j0sh** [comment]
A long-term fix might be to briefly persist the data server-side using the original connection ID as the key, and have the client reconnect using the original ID. But that would be a significant and somewhat fraught change.
